### DISCUSSION TITLE
[General] Error on listening to events in mainnet, could not coalesce error (error={ "code": -32602 }

### DISCUSSION THREAD

**Discussion by waelsy123** [zkSync-Community-Hub/zksync-developers#410] [General]
### Environment

Mainnet

### Provide a brief description of the functionality you're trying to implement.

I'm trying to listen to events emitted by smart contracts on zkSync era Mainnet 

### What is the specific issue or error you're encountering?

The issue happens only in mainnet. On testnet it works properly. 

### Can you share the error messages or logs you're receiving, if any?

@TODO Error: could not coalesce error (error={ "code": -32602, "message": "Filter not found" }, payload={ "id": 1228, "jsonrpc": "2.0", "method": "eth_getFilterChanges", "params": [ "0x14fe6bb558ee24ae0f26531c9bf455473e1ff60cc0ba241256408cf15c71507d" ] }, code=UNKNOWN_ERROR, version=6.10.0)
    at makeError (/greater_example/node_modules/ethers/src.ts/utils/errors.ts:694:21)
    at Provider.getRpcError (/greater_example/node_modules/ethers/src.ts/providers/provider-jsonrpc.ts:1049:25)
    at /greater_example/node_modules/ethers/src.ts/providers/provider-jsonrpc.ts:563:45
    at processTicksAndRejections (node:internal/process/task_queues:95:5) {
  code: 'UNKNOWN_ERROR',
  error: { code: -32602, message: 'Filter not found' },
  payload: {
    method: 'eth_getFilterChanges',
    params: [
      '0x14fe6bb558ee24ae0f26531c9bf455473e1ff60cc0ba241256408cf15c71507d'
    ],
    id: 1228,
    jsonrpc: '2.0'
  },
  shortMessage: 'could not coalesce error'
}

### Have you made any recent changes to the contract before encountering this issue?

no

### Are there any external libraries or contracts that your contract interacts with?

I'm using hardhat with ethers library

### Can you provide the relevant portions of your contract code where the issue is occurring?

const provider = await getProvider();
const zkETH = await ethers.getContractAt("zkETH", process.env.zkETHContract!, owner) as any;
const owner = new Wallet(process.env.DEPLOYER_PRIVATE_KEY!, provider);
  
 zkETH.on('Withdrawal', (token: string, to: string, amount: number, event: any) => {
    console.log(`Withdrawal: Token - ${token}, To - ${to}, Amount - ${amount.toString()}`);
  }).catch(e => {
    console.log("🚀 ~ zkETH.on('Withdrawal') ~ e:", e)
  });

### Have you tried to isolate the problem, and if so, what were the results?

It happen with all smart contracts for me

### What steps have you already taken to try to resolve the issue?

let me know your thoughts

**Comment by dutterbutter** [comment]
Hey @waelsy123 can you please provide a minimal example or access to the code in order to reproduce?

**Comment by dutterbutter** [comment]
Feel free to re-open if the issue remains. Closing due to staleness.

**Comment by fedealconada** [comment]
I'm experiencing this as well... seems like events are being detected but still the error is constantly logged...

**Reply by fedealconada** [reply]
@waelsy123 have you sorted this out?
